Changetracking implementation (#4843)
ElektraInitiative/libelektra: doc/news/_preparation_next_release.md

## Highlights

- New Changetracking API
- <<HIGHLIGHT>>
- <<HIGHLIGHT>>

### New Changetracking API

We've created a new KeySet diffing and changetracking API for both internal and external use.
Several plugins already had their own implementations to detect changed made to the key database.
Unfortunately, every implementation did something different and results varied.
With this new release every plugin now uses the same shared implementation!

> If you want to use the new API yourself, take a look at [the tutorial](../tutorials/changetracking.md)!

Reviewer: ```suggestion
To try the amazing new API yourself, take a look at [the tutorial](../tutorials/changetracking.md)!
```

Elektra's internal code now also uses this new API to detect which backends to execute.
This can lead to better performance in I/O bound cases, where previously certain keys would have been detected as changed when they weren't.

We ran some memory benchmarks and found a slightly increased memory usage on a stock instance of Elektra.
If you're using many plugins that do changetracking, the overhead will decrease.

> | Number of Keys | Old Implementation (bytes) | New Implementation (bytes) | Memory Increase (%) |

Author: Yes!

| -------------: | -------------------------: | -------------------------: | ------------------: |
|             50 |               225792 bytes |               229580 bytes |              1,68 % |
|            500 |               383180 bytes |               411238 bytes |              7,32 % |
|           5000 |              1992294 bytes |              2306867 bytes |             15,79 % |
|          50000 |             18245222 bytes |             21181235 bytes |             16,09 % |

> |         500000 |            178782208 bytes |            207827763 bytes |             16,25 % |

Reviewer: Interesting non-linearity. In later research you might look into it more closely.

Author: Yeah I've looked very closely into it. It's because at a certain number of keys they become the dominant factor of memory consumption. So at 50 keys the overall memory consumption of Elektra is not really affected by the keys. But at > 5000 the other memory consumption doesn't matter much.
